src/lib/types.ts

```diff
@@ -14,12 +13,25 @@ export type FileLoaderData = {
 }
 
 export type ProjectWithEntryPointMetadata = FileEntry & {
-  entrypointMetadata: Metadata
+  entrypointMetadata: FileInfo
 }
 export type HomeLoaderData = {
   projects: ProjectWithEntryPointMetadata[]
 }
 
+// From https://github.com/tauri-apps/tauri/blob/1.x/tooling/api/src/fs.ts#L159
+// Removed from tauri v2
+export interface FileEntry {
+  path: string
+  /**
+   * Name of the directory/file
+   * can be null if the path terminates with `..`
+   */
+  name?: string
+  /** Children of this entry if it's a directory; null otherwise */
+  children?: FileEntry[]
+}
+
```

Comment:
Is there a better type I should use from `@tauri-apps/plugin-fs`? I'd rather we use their type if they have one exposed, otherwise couldn't we have a mismatch between our expected shape of the data and the actually-returned data from their read function. I guess it's not a concern since we're shimming our own `read_dir_recursive` function from V1 everywhere?

Reply:
@franknoirot There is a new type yes, but it's not recursive-able. So unless we find a way to rewrite the filesystem reading logic we have to use the non-recursive `readDir` we can't really use it I think

It's `DiskEntry`, defined at https://github.com/tauri-apps/plugins-workspace/blob/v2/plugins/fs/guest-js/index.ts#L625-L641